docs/enhancements/dual-stack-vips.md

```diff
@@ -0,0 +1,142 @@
+see-also:
+  - https://github.com/openshift/enhancements/blob/master/enhancements/network/on-prem-dual-stack-vips.md
+---
+
+# Dual-stack VIPs
+
+## Summary
+
+Originally Assisted Installer supported only single-stack IPv4 or IPv6 deployments. When dual-stack support was added, we carried over the VIP configuration that only allowed for a single VIP to be specified.
+
+The initial rationale was that in dual-stack clusters every required communication can happen over IPv4 and only the workload running in the cluster may need IPv6 connectivity.
+
+Given this is no longer the case and that OCP for the baremetal platform supports dual-stack VIPs, this design is proposing the addition of a Cluster property allowing configuration of the aforementioned feature.
+
+## Motivation
+
+The current implementation of dual-stack support assumes that everything in the cluster should be reachable via IPv4 and that IPv6 is an addition for the workload running in the cluster. This assumption makes it impossible to interact with such a cluster using IPv6-only client, i.e. makes it impossible for IPv6-only Hub to manage a dual-stack Spoke.
+
+### Goals
+
+* Allow configuring both IPv4 and IPv6 VIP for dual-stack clusters.
+* Allow configuring only IPv4 VIPs for dual-stack clusters.
+
+### Non-goals
+
+* Allow configuring more than single IPv4 and single IPv6 VIP. This is still not supported by the OCP. MetalLB is a better solution to creating arbitrary loadbalancers.
+* Allow configuring IPv6 as the primary networking stack. This is still not supported by the OCP. For dual-stack deployments the IPv4 stack is always to be the primary.
+* Allow DHCPv6 allocation of IPv6 VIPs. This is because DHCPv6 is not the model way of deploying IPv6 networks as well as that we intentionally decided not to offer IPv4 DHCP allocation for dual-stack clusters.
+* Allow VIP(s) to be placed outside of subnet(s) defined as machine network(s).
+
+## Proposal
+
+### User Stories
+
+* As an operator of the IPv6-only Hub cluster I want to deploy dual-stack Spoke cluster.
+
+* As an user of a dual stack OCP cluster I want to run IPv6-only applications that need access to the API and/or Ingress VIPs.
+
+### API Extensions
+
+New fields will have to be added to the Cluster object to persist the specified VIPs.
+
+### Implementation Details
+
+Our change can be simply described as "making a singular field plural". For this, we will follow the [Kubernetes API change guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#making-a-singular-field-plural).
+
+#### Phase 1
+
+`cluster-create-params`, `v2-cluster-update-params` and `cluster` will get new fields
+
+* `api_vips` of type `array`
+* `ingress_vips` of type `array`
+
+Respective AgentClusterInstall CR will also get those new fields.
+
+The database schema will be updated by creating new tables `api_vips` and `ingress_vips` with the following columns
```

Comment:
Perhaps the document should mention that this will be done via DB migration.

Reply:
Of course, just added it